exporter/jaegerexporter/exporter.go

```diff
@@ -40,21 +45,30 @@ func newTraceExporter(cfg *Config, logger *zap.Logger) (component.TracesExporter
 		return nil, err
 	}
 
-	client, err := grpc.Dial(cfg.GRPCClientSettings.Endpoint, opts...)
+	conn, err := grpc.Dial(cfg.GRPCClientSettings.Endpoint, opts...)
 	if err != nil {
 		return nil, err
 	}
 
-	collectorServiceClient := jaegerproto.NewCollectorServiceClient(client)
+	collectorServiceClient := jaegerproto.NewCollectorServiceClient(conn)
 	s := &protoGRPCSender{
+		name:         cfg.NameVal,
 		logger:       logger,
 		client:       collectorServiceClient,
 		metadata:     metadata.New(cfg.GRPCClientSettings.Headers),
 		waitForReady: cfg.WaitForReady,
+
+		conn:                      conn,
+		connStateReporterInterval: time.Second,
+
+		stopCh: make(chan (struct{})),
 	}
+	s.AddStateChangeCallback(s.onStateChange)
```

Comment:
> Is there ever more than one callback needed?

No, perhaps in tests only. In any case, I think it's a good general pattern to allow multiple callbacks to exist. I'm happy to make it a simple property if you prefer.

> It is not clear why this gets a special treatment while other fields are initialized directly.

It's because it needs to refer the instance `s`.